tests/framework/pkg/config/config.go

```diff
@@ -93,18 +95,24 @@ func LoadAndUpdateConfig(key string, config any, updateFunc func()) {
 }
 
 // WriteConfig writes a CATTLE_TEST_CONFIG config file when one is not previously written.
-func WriteConfig(key string, config interface{}) {
+func WriteConfig(key string, config interface{}) error {
```

Comment:
I'd prefer to return an error over panicking here.

Reply:
Do you have a specific reason for this? Ideally we kept the config file stuff as panics simply because if these steps failed then we should panic since the framework needs to config files to be able to work. I'm open to changing this, just wanted an explanation.